packages/office-ui-fabric-react/src/utilities/decorators/withResponsiveMode.tsx

```diff
@@ -46,7 +45,7 @@ export function withResponsiveMode<TProps extends { responsiveMode?: ResponsiveM
       this._updateComposedComponentRef = this._updateComposedComponentRef.bind(this);
 
       this.state = {
-        responsiveMode: _defaultMode || _lastMode || ResponsiveMode.large
+        responsiveMode: this._getResponsiveMode()
```

Comment:
Is there any scenario where any of the values used in getResponsiveMode won't be available yet?

Also, you may want to remove the onResize call in componentDidMount.